### PR TITLE
Data model changes for v2

### PR DESCRIPTION
**Pull request by stark-3k**
- [x] Add `proposed_by_contributor_id` to `Allocation`
- [x] Add `status` to `Allocation`
- [x] Add `expected_budget_currency` to `Project`
- [x] Add `is_public` to `Project`
- [x] Add `project_id` to `Payment` and allow `client_id` to be `NULL`
- [x] Add `Phase` table
- [x] Add `Wallet` table

[Reference](https://www.figma.com/file/qgGWXmprU7vTv7guzWzvML/Project-Trinary)

### CODE REVIEW

**Comment by sofiaromorales** [comment]
@rishkwal the pr seems ok but you still need to add the DB changes to the models

File: api/migrations/20221110035638-addIsPrivateToProject.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    down: async (queryInterface, Sequelize) => {
+        return queryInterface.sequelize.transaction(t => {
+            return Promise.all([
+                queryInterface.removeColumn('Projects', 'is_public',
+                    { transaction: t })
+            ])
+        })
+    }
+};
```
Comment (sofiaromorales) [inline api/migrations/20221110035638-addIsPrivateToProject.js]: I already made this change here #720

File: api/models/Phase.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            type: DataTypes.STRING,
+            allowNull: false
+        },
+        color: {
+            type: DataTypes.INTEGER,
+            allowNull: false
+        }
+    }, {
+        sequelize,
+        modelName: 'Phase',
```
Comment (sofiaromorales) [inline api/models/Phase.js:47]:
```js
sequelize,
modelName: 'Phase',
createdAt: 'created_at',
updatedAt: 'updated_at'
```

The same applies to the other models

**Review by sofiaromorales — CHANGES_REQUESTED** [review]

**Comment by sofiaromorales** [comment]
@rishkwal I'm blocked by some of these changes, pls update the pr so it can be merged ASAP

**Comment by stark-3k** [comment]
> @rishkwal I'm blocked by some of these changes, pls update the pr so it can be merged ASAP

@sofiaromorales I've made the changes as discussed. Let me know if I need to make any other change.

**Review by sofiaromorales — APPROVED** [review]

**Pull request merged**